content/blog/2020-08-27-august-20-community-gems.md

```diff
@@ -0,0 +1,179 @@
+commentsUrl: https://discuss.dvc.org/t/august-20-community-gems/477
+tags:
+  - Discord
+  - Gems
+  - CML
+  - hyperparameters
+  - LFS
+  - pipeline
+---
+
+Here are some of our top Q&A's from around the community. With the launch of
+[CML](https://cml.dev) earlier in the month, we've got some new ground to cover!
+
+## DVC questions
+
+### [Q: What's the relationship between your DVC remote and cache? If I have an external cache, do I really need a DVC remote](https://discordapp.com/channels/485586884165107732/563406153334128681/747588572479094866)
+
+You can think of your DVC remote similar to your Git remote- it's a place to
+backup and share artifacts, like datasets and models, and it gives you methods
+to push and pull those artifacts to and from your team.
+
+By default, `.dvc/cache` is also similar to your `.git` cache. They're both
+_local_ caches that optimize access to files (data and models for DVC cache, and
+code for Git cache).
+
+Your DVC remote is a superset of `.dvc/cache`- everything in your cache is a
+copy of something in your remote (though there may be files in your DVC remote
+that are not in your cache, if you have never attempted to `push` or `pull` them
+locally).
+
+In theory, if you are using an
+[external cache](https://dvc.org/doc/command-reference/destroy#example-external-cache-directory)
+(such as an S3 bucket), and all your projects and all your teammates use that
+external cache, and you _know_ that the storage is highly reliable, you don't
+need to also have a DVC remote. If you have any doubts about access to your
+external cache or its reliability, we'd recommend also keeping a remote.
+
+### [Q: One of my files is an output of a DVC pipeline, and I want to track this file with Git and store it in my Git repository since it isn't very big. How can I make this work?](https://discordapp.com/channels/485586884165107732/563406153334128681/732308317627613235)
+
+If a file (call it `myfile`) is the output of a DVC pipeline, DVC takes several
+steps to remove it from Git tracking. If you remove it from the `.gitignore` and
+do `git add myfile`, you'll get an error when you try to reproduce your pipeline
+with `dvc repro` (`myfile is already trakced by SCM`).
+
+To get around this, you need to specify that DVC should not cache the outputs of
+your pipeline]. When you declare a pipeline stage with `dvc run`, instead of
+using the `-o` flag, use `-O`. This is shorthand for `--outs-no-cache` ([see
+docs for `dvc run` and its flags
+[here](https://dvc.org/doc/command-reference/run#options)).
+
+In your `dvc.yaml` file, if you prefer to manually edit, you'll add a field for
+`cache`:
+
+```yaml
+outs:
+  - myfile:
+      cache: false
+```
+
+### [Q: Can I change my `params.yaml` file to a `.json`?](https://discordapp.com/channels/485586884165107732/563406153334128681/730614265051873370)
+
+Yes, this is straightforward- you change your `params.yaml` to `params.json` in
+your workspace, and then use it in `dvc run`:
```

Comment:
... or in `dvc.yaml` (if stages already created)

Reply:
added